### PR TITLE
storage: remote {copy,move} custom storage volumes

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: client/lxd_storage_volumes.go [hunk 1]
```diff
@@ -94,16 +94,143 @@ func (r *ProtocolLXD) CreateStoragePoolVolume(pool string, volume api.StorageVol
 	return nil
 }
 
+// MigrateStoragePoolVolume requests that LXD prepares for a storage volume migration
+func (r *ProtocolLXD) MigrateStoragePoolVolume(pool string, volume api.StorageVolumePost) (Operation, error) {
```
Comment (stgraber) [inline client/lxd_storage_volumes.go]: API extension check?

File: client/lxd_storage_volumes.go [hunk 2]
```diff
@@ -114,24 +241,150 @@ func (r *ProtocolLXD) CopyStoragePoolVolume(pool string, source ContainerServer,
...
+		go func() {
+			rop.err = rop.targetOp.Wait()
+			close(rop.chDone)
+		}()
+
+		return &rop, nil
+	}
+
+	if !r.HasExtension("storage_api_remote_volume_handling") {
+		return nil, fmt.Errorf("The server is missing the required \"storage_api_local_volume_handling\" API extension")
```
Comment (stgraber) [inline client/lxd_storage_volumes.go]: Wrong extension name in the error

**Comment by stgraber** [comment]
Looks pretty good to me, just a few minor nits

File: doc/api-extensions.md [hunk 3]
```diff
@@ -446,3 +446,6 @@ The following existing endpoints have been modified:
 
 ## event_lifecycle
 This adds a new `lifecycle` message type to the events API.
+
+## storage\_api\_remote\_volume\_handling
+This add the ability to copy and move custom storage volumes remotely.
```
Comment (brauner) [inline doc/api-extensions.md]: s/add/adds/

File: doc/rest-api.md [hunk 4]
```diff
@@ -2332,6 +2347,26 @@ Input:
...
+    }
+
+The migration does not actually start until someone (i.e. another lxd instance)
+connects to all the websockets and begins negotiation with the source.
+
+Output in metadata section (for migration):
+
+    {
+        "control": "secret1",       # Migration control socket
+        "fs": "secret3"             # Filesystem transfer socket
```
Comment (brauner) [inline doc/rest-api.md]: s/secret3/secret2/

File: doc/api-extensions.md (hunk 3 above)
Comment (brauner) [reply doc/api-extensions.md]: addressed

File: doc/rest-api.md (hunk 4 above)
Comment (brauner) [reply doc/rest-api.md]: addressed

**Comment by brauner** [comment]
jenkins: test this please

**Pull request merged**